src/me/mrCookieSlime/Slimefun/listeners/BackpackListener.java

```diff
@@ -82,62 +83,61 @@ else if (e.getCurrentItem() != null && e.getCurrentItem().getType() != Material.
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.BACKPACK_MEDIUM, false)) {
-				openBackpack(SlimefunItems.BACKPACK_MEDIUM, 18, e, p, item);
-			}
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.BACKPACK_LARGE, false)) {
-				openBackpack(SlimefunItems.BACKPACK_LARGE, 27, e, p, item);
-			}
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.WOVEN_BACKPACK, false)) {
-				openBackpack(SlimefunItems.WOVEN_BACKPACK, 36, e, p, item);
-			}
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.GILDED_BACKPACK, false)) {
-				openBackpack(SlimefunItems.GILDED_BACKPACK, 45, e, p, item);
-			}
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.RADIANT_BACKPACK, false)) {
-				openBackpack(SlimefunItems.RADIANT_BACKPACK, 54, e, p, item);
-			}
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.BOUND_BACKPACK, false)) {
-				openBackpack(SlimefunItems.BOUND_BACKPACK, 36, e, p, item);
-			}
-			else if (SlimefunManager.isItemSimiliar(item, SlimefunItems.COOLER, false)) {
-				openBackpack(SlimefunItems.COOLER, 27, e, p, item);
-			}
+			ItemStack item = e.getItem();
+			SlimefunItem sfItem = SlimefunItem.getByItem(item);
+			if (sfItem == null) return;
+
+			// Surely this can be even simpler?
+			if (sfItem.isItem(SlimefunItems.BACKPACK_SMALL))
+				openBackpack(SlimefunItems.BACKPACK_SMALL, 9, e, p);
+			else if (sfItem.isItem(SlimefunItems.BACKPACK_MEDIUM))
+				openBackpack(SlimefunItems.BACKPACK_MEDIUM, 18, e, p);
+			else if (sfItem.isItem(SlimefunItems.BACKPACK_LARGE))
+				openBackpack(SlimefunItems.BACKPACK_LARGE, 27, e, p);
+			else if (sfItem.isItem(SlimefunItems.WOVEN_BACKPACK))
+				openBackpack(SlimefunItems.WOVEN_BACKPACK, 36, e, p);
+			else if (sfItem.isItem(SlimefunItems.GILDED_BACKPACK))
+				openBackpack(SlimefunItems.GILDED_BACKPACK, 45, e, p);
+			else if (sfItem.isItem(SlimefunItems.RADIANT_BACKPACK))
+				openBackpack(SlimefunItems.RADIANT_BACKPACK, 54, e, p);
+			else if (sfItem.isItem(SlimefunItems.BOUND_BACKPACK))
+				openBackpack(SlimefunItems.BOUND_BACKPACK, 36, e, p);
+			else if (sfItem.isItem(SlimefunItems.COOLER))
+				openBackpack(SlimefunItems.COOLER, 27, e, p);
 		}
 	}
 
-	private void openBackpack(ItemStack sfItem, int size, PlayerInteractEvent e, Player p, ItemStack item) {
+	private void openBackpack(ItemStack item, int size, PlayerInteractEvent e, Player p) {
 		e.setCancelled(true);
-		
+
 		if (item.getAmount() == 1) {
-			if (Slimefun.hasUnlocked(p, sfItem, true)) {
-				Slimefun.getLocal().sendMessage(p, "messages.opening-backpack");
-				PlayerProfile.get(p, profile -> {
-                    for (int line = 0; line < item.getItemMeta().getLore().size(); line++) {
-                        if (item.getItemMeta().getLore().get(line).equals(ChatColor.translateAlternateColorCodes('&', "&7ID: <ID>"))) {
-                            BackpackInventory backpack = profile.createBackpack(size);
+			if (Slimefun.hasUnlocked(p, item, true)) {
```

Comment:
Could be improved if you use the SlimefunItem variant here, otherwise another call to SlimefunItem.getByItem() is made.
Therefore you should pass the newly added SlimefunItem as a parameter.